core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -794,22 +793,23 @@ class ReplicaManagerTest {
 
       // We receive one valid request from the follower and replica state is updated
       var successfulFetch: Option[FetchPartitionData] = None
-      def callback(response: Seq[(TopicPartition, FetchPartitionData)]): Unit = {
-        successfulFetch = response.headOption.filter { case (topicPartition, _) => topicPartition == tp }.map { case (_, data) => data }
+      def callback(response: Seq[(TopicIdPartition, FetchPartitionData)]): Unit = {
+        // Check the topic partition only since we are reusing this callback on different TopicIdPartitions.
+        successfulFetch = response.headOption.filter { case (topicIdPartition, _) => topicIdPartition.topicPartition == tidp.topicPartition }.map { case (_, data) => data }
```

Comment:
This is not ideal. Could we validate that the topic id is correct as well?

Reply:
STILL TODO for Friday

Reply:
So I can write a separate callback for each one that checks the ID.